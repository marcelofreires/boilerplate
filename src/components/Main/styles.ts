import styled from 'styled-components'

export const BACKGROUND_COLOR = '#1a1b28'
export const TITLE_COLOR = '#00fe57'

export const Container = styled.main`
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: ${BACKGROUND_COLOR};
`

export const Title = styled.h1`
  color: ${TITLE_COLOR};
  font-size: 3rem;
  font-weight: 700;
`
