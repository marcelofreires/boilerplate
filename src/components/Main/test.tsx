import { queryByText, render } from '@testing-library/react'

import { Main } from './'
import { BACKGROUND_COLOR, TITLE_COLOR } from './styles'

describe('<Main />', () => {
  it('should render the component', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(queryByText(container, 'Next.js boilerplate')).toBeInTheDocument()
  })

  it('should render the background color correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({
      'background-color': BACKGROUND_COLOR
    })
  })

  it('should render the title color correctly', () => {
    const { container } = render(<Main />)

    expect(queryByText(container, 'Next.js boilerplate')).toHaveStyle({
      color: TITLE_COLOR
    })
  })
})
