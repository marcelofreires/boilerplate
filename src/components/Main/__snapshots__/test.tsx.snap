// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the component 1`] = `
.c0 {
  width: 100%;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #1a1b28;
}

.c1 {
  color: #00fe57;
  font-size: 3rem;
  font-weight: 700;
}

<main
  class="c0"
  id="main"
>
  <h1
    class="c1"
    id="title"
  >
    Next.js boilerplate
  </h1>
</main>
`
